React 15.4 breaks hot-reloading

From my attempts to start the Hello World example:

```
ERROR in /Users/dwade/foss/react-server/~/react-dom/lib/ReactDebugTool.js
Module not found: Error: Cannot resolve 'file' or 'directory' /Users/dwade/foss/react-server/packages/react-server-examples/hello-world/node_modules/react/lib/ReactComponentTreeHook in /Users/dwade/foss/react-server/node_modules/react-dom/lib
 @ /Users/dwade/foss/react-server/~/react-dom/lib/ReactDebugTool.js 16:29-72

ERROR in /Users/dwade/foss/react-server/~/react-dom/lib/checkReactTypeSpec.js
Module not found: Error: Cannot resolve 'file' or 'directory' /Users/dwade/foss/react-server/packages/react-server-examples/hello-world/node_modules/react/lib/ReactComponentTreeHook in /Users/dwade/foss/react-server/node_modules/react-dom/lib
 @ /Users/dwade/foss/react-server/~/react-dom/lib/checkReactTypeSpec.js 29:27-70 71:37-80
```
looks like its [an outdated version of react-hot-loader](https://github.com/gaearon/react-hot-loader/issues/417#issuecomment-260989221)

not sure why my initial testing of https://github.com/redfin/react-server/pull/780 didn't catch this 😞 but I don't think its worth reverting, since the real problem is that react 15 doesn't play nice with react-server

--- a/packages/react-server-cli/package.json
+++ b/packages/react-server-cli/package.json
@@ -50,15 +50,16 @@
   },
   "devDependencies": {
     "ava": "^0.16.0",
+    "babel-eslint": "^7.0.0",
     "babel-plugin-add-module-exports": "^0.2.1",
     "babel-preset-react-server": "^0.4.10",
+    "eslint-plugin-react": "^6.4.1",
     "gulp": "^3.9.1",
     "gulp-babel": "^6.1.2",
     "gulp-eslint": "^3.0.1",
     "nsp": "^2.6.2",
+    "react-hot-loader": "^1.3.1",
     "react-server-gulp-module-tagger": "^0.4.10",
-    "rimraf": "^2.5.4",
-    "babel-eslint": "^7.0.0",
-    "eslint-plugin-react": "^6.4.1"
+    "rimraf": "^2.5.4"
   }
 }

